dl4j-cv-labs/src/main/java/ai/certifai/utilities/VocLabelProvider.java

```diff
@@ -0,0 +1,115 @@
+package ai.certifai.utilities;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.NonNull;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.datavec.image.recordreader.objdetect.ImageObject;
+import org.datavec.image.recordreader.objdetect.ImageObjectLabelProvider;
+
+public class VocLabelProvider implements ImageObjectLabelProvider {
+    private static final String OBJECT_START_TAG = "<object>";
+    private static final String OBJECT_END_TAG = "</object>";
+    private static final String NAME_TAG = "<name>";
+    private static final String XMIN_TAG = "<xmin>";
+    private static final String YMIN_TAG = "<ymin>";
+    private static final String XMAX_TAG = "<xmax>";
+    private static final String YMAX_TAG = "<ymax>";
+    private String annotationsDir;
+
+    public VocLabelProvider(@NonNull String baseDirectory) {
+        if (baseDirectory == null) {
+            throw new NullPointerException("baseDirectory is marked non-null but is null");
+        } else {
+            this.annotationsDir = FilenameUtils.concat(baseDirectory, "Annotations");
+            if (!(new File(this.annotationsDir)).exists()) {
+                throw new IllegalStateException("Annotations directory does not exist. Annotation files should be present at baseDirectory/Annotations/nnnnnn.xml. Expected location: " + this.annotationsDir);
+            }
+        }
+    }
+
+    public List<ImageObject> getImageObjectsForPath(String path) {
+        int idx = path.lastIndexOf(47);
```

Comment:
not an issue but any reason you use unicode int instead of the more easily understood char '/' ? Same question goes for l38, l39

Reply:
no idea, I just followed the convention of what the original writer did..
thought would be somewhat slightly better performance...
after looking at some resources from internet, it seems negligible. Will make the change for easy reading as well..